### PR TITLE
Generalise _PORT_RE to cope with LAG

### PR DESCRIPTION
**Pull request by bmerry**

### CODE REVIEW

**Comment by bmerry** [comment]
This is worth reviewing one commit at a time, since it does two different things.

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,30 @@
 [build-system]
-requires = ["setuptools", "wheel", "katversion"]
+requires = ["setuptools", "katversion"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "switch_exporter"
+dynamic = ["version"]
+authors = [{name = "MeerKAT SDP team", email = "sdpdev+switch_exporter@ska.ac.za"}]
```
Comment (james-smith-za) [inline pyproject.toml:8]: You might want to update this email alias too.

File: src/switch_exporter/switch.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 
 logger = logging.getLogger(__name__)
-_PORT_RE = re.compile(r'^Eth([^:]*):?$')
+_PORT_RE = re.compile(r'^Eth([^ :]*)(?: \(.*\))?:?$')
```
Comment (james-smith-za) [inline src/switch_exporter/switch.py:16]: The second capture group which you're just discarding here isn't just an arbitrary set of characters, it'll be something like `\(Po[\d]+\)`.

I'm not sure if this information will be useful, if we'd like to consider the LAGs separately for instance.

I'm not currently aware of any circumstances where anything other than `Po[digit]` will be there so I am not sure of the value of trying to be more specific.

**Review by james-smith-za — COMMENTED** [review]
Just s/ska/sarao/ I think is necessary.

The comment on the regex is more of an observation than a change request, I'd obviously prefer it to be a bit more specific but this is a loosely-held opinion, it might not be worth the effort, or there might be good reason to do it your way rather than mind.

File: src/switch_exporter/switch.py (hunk 2 above)
Comment (bmerry) [reply src/switch_exporter/switch.py:16]: I was also unsure about how specific to try to be. Too specific and we're just going to have to revisit when some other string turns up; too generic and we might mis-identify something. Without any examples of what else might occur it's hard to make an informed decision, but my gut feel is that if something else appears between parentheses, we probably want to ignore it too.

File: pyproject.toml (hunk 1 above)
Comment (bmerry) [reply pyproject.toml:8]: Dang, thought I had. Fixed now.

**Pull request merged**